plugins/notion/src/FieldMapping.tsx

```diff
@@ -0,0 +1,323 @@
+                if (!fieldInfo.allowedTypes.includes(type)) return fieldInfo
+                return { ...fieldInfo, type }
+            })
+            return updatedFieldInfo
+        })
+    }
+
+    const toggleFieldIgnoredState = (fieldId: string) => {
+        setIgnoredFieldIds(previousIgnoredFieldIds => {
+            const updatedIgnoredFieldIds = new Set(previousIgnoredFieldIds)
+
+            if (updatedIgnoredFieldIds.has(fieldId)) {
+                updatedIgnoredFieldIds.delete(fieldId)
+            } else {
+                updatedIgnoredFieldIds.add(fieldId)
+            }
+
+            return updatedIgnoredFieldIds
+        })
+    }
+
+    const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+
+        if (!selectedSlugFieldId) {
+            // This can't happen because the form will not submit if no slug field is selected
+            // but TypeScript can't infer that.
+            console.error("There is no slug field selected. Sync will not be performed")
+            framer.notify("Please select a slug field before importing.", { variant: "warning" })
+            return
+        }
+
+        try {
+            setStatus("syncing-collection")
+
+            const fields = await fieldsInfoToCollectionFields(fieldsInfo)
+            const fieldsToSync = fields.filter(field => !ignoredFieldIds.has(field.id))
+            const slugField = fields.find(field => field.id === selectedSlugFieldId)
+
+            if (!slugField) {
+                framer.notify("Selected slug field not found. Sync will not be performed.", { variant: "error" })
+                return
+            }
+
+            await syncCollection(collection, dataSource, fieldsToSync, slugField, ignoredFieldIds, previousLastSynced)
+            await framer.closePlugin("Synchronization successful", { variant: "success" })
+        } catch (error) {
+            console.error(error)
+            framer.notify(`Failed to sync collection “${dataSource.id}”. Check the logs for more details.`, {
+                variant: "error",
+            })
+        } finally {
+            setStatus("mapping-fields")
+        }
+    }
+
+    if (isLoadingFields) {
+        return (
+            <main className="loading">
+                <div className="framer-spinner" />
+            </main>
+        )
+    }
+
+    return (
+        <main className="framer-hide-scrollbar mapping">
+            <hr className="sticky-divider" />
+            <form onSubmit={handleSubmit}>
+                <label className="slug-field" htmlFor="slugField">
+                    <div className="heading-row">
+                        <span>Slug Field</span>
+                        {database?.url && (
+                            <a href={database.url} target="_blank" className="heading-link">
+                                View in Notion
+                            </a>
+                        )}
```

Comment:
This is a nice idea, but it needs a better design, IMO.

It would be best to add an issue, because we should try and keep this PR as focused as possible on using the CMS Starter structure 🙏 

<img width="450" alt="image" src="https://github.com/user-attachments/assets/d190eca3-d040-4708-a6ae-705c3c1f7f65" />